crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -707,6 +707,11 @@ pub fn comparator_operand_type_rules(
             // We could check if all elements of all arrays are comptime but I am lazy
             Ok(Bool(Comptime::No(Some(op.location.span))))
         }
-        (lhs, rhs) => Err(format!("Unsupported types for comparison: {} and {}", lhs, rhs)),
+        (lhs, rhs) => {
+            if lhs == rhs {
```

Comment:
Comparison operators are not valid for all types (e.g. `myfn1 < myfn2` is invalid). If any other types are required, we should add them individually.
Since comparisons are not valid for every type, a proper sort function would likely require higher order functions to be able to pass in the appropriate `<` instance.

Reply:
Ok, I have enabled the comparison for generics only, because we do not know at this stage here if it is valid or not. If the types do not support this in fine, the user will get an error later on.